daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/parsers/SequenceParserBases.scala

```diff
@@ -332,6 +331,8 @@ abstract class SequenceParserBase(
         case other => Assert.invariantFailed("Unexpected parse attempt status: " + other)
       }
 
+    if (hasPoU) pstate.popPointOfUncertainty
```

Comment:
This looks like the unwinding bug fix. Yes?

Reply:
Yes, we were popping the POU stack before calling PState.reset() and thus the PState.reset() was resetting variables outside of the POU.